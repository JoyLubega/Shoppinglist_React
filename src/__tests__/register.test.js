import React from 'react';
import { mount,shallow } from 'enzyme';
import { StaticRouter } from 'react-router';
import Register from '../components/authentication/register.js';
import renderer from 'react-test-renderer';
import moxios from 'moxios';
import sinon from 'sinon';

const baseURL = "http://127.0.0.1:5000";
const axios = require('axios');
const MockAdapter = require('axios-mock-adapter');

// This sets the mock adapter on the default instance
const mock = new MockAdapter(axios);
// Mock Login POST request to /login
mock.onPost(`${baseURL}/auth/register`).reply(200, {
  users: [
    { status: 'success', token: 1 },
  ],
});

global.localStorage = {
  getItem: () => {},
  setItem: () => {},
};

describe('Component: Register', () => {
  const wrapper = shallow(<Register />)
  const routerComponent = mount(
    <StaticRouter location="register" context={{}}>
      <Register/>
    </StaticRouter>,
  );
  const RegisterComponent = routerComponent.find('Register');
  const nameInput = RegisterComponent.find('#name');
  const emailInput = RegisterComponent.find('#email');
  const passwordInput = RegisterComponent.find('#password');

  it('Displays Register', () => {
    const rendered = renderer.create(
      <StaticRouter location="register" context={{}}>
        <Register />
      </StaticRouter>,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
  it('should render form correctly', () => {
        // check register component contain one <from/>
        const LoginForm = wrapper.find("form");
        expect(LoginForm).toHaveLength(1);

});

    it('should render form inputs', () => {
            expect(wrapper.find('#name').length).toEqual(1);
            expect(wrapper.find('#email').length).toEqual(1);
            expect(wrapper.find('#password').length).toEqual(1);
    });


    it('input should respond to change event and change the state', () => {
          wrapper.find('#email').simulate('change', { target: { name: 'email', value: 'admin@gmail.com' } });
          expect(wrapper.state('email')).toEqual('admin@gmail.com')
      });

  it('Registers a user', () => {
    const registerButton = RegisterComponent.find('[type="submit"]');
    registerButton.simulate('submit');
  });
});
