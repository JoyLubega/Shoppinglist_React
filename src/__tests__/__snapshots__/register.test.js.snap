// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Register Displays Register 1`] = `
<div
  className=" Register_container container-fluid "
>
  <div
    className="col-md-4 col-md-offset-4 col-sm-9 auth-box"
  >
    <div
      className="charis navbar"
    >
      Charis
    </div>
    <div
      className="slist"
    >
      Shopping List
    </div>
    <div
      className="register-div row"
    >
      <div
        className=" log_title"
      >
        Register
      </div>
      <form
        className=" login_form col s12"
        onSubmit={[Function]}
      >
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <i
              className="material-icons prefix"
            >
              account_circle
            </i>
            <input
              className="validate"
              id="name"
              name="name"
              onChange={[Function]}
              type="text"
            />
            <label>
              Name
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <i
              className="material-icons prefix"
            >
              email
            </i>
            <input
              className="validate"
              id="email"
              name="email"
              onChange={[Function]}
              type="email"
            />
            <label>
              Email
            </label>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="input-field col s12"
          >
            <i
              className="material-icons prefix"
            >
              lock
            </i>
            <input
              className="validate"
              id="password"
              name="password"
              onChange={[Function]}
              type="password"
            />
            <label>
              Password
            </label>
          </div>
        </div>
        <button
          className="send  btn waves-effect waves-light"
          name="action"
          type="submit"
        >
          Submit
          <i
            className="material-icons right"
          >
            send
          </i>
        </button>
        <div
          className="link_register row"
        >
          <a
            className=" reg btn btn-outline-info"
            href="/login"
          >
            Login
          </a>
        </div>
      </form>
    </div>
  </div>
  <div />
</div>
`;
